### ISSUE TITLE
Transpose op doesn't work when dimensions are not divisible by tile

### ISSUE THREAD

**Issue by dgolubovicTT** [tenstorrent/tt-metal#13005]
### **Describe the bug**
for tensor shape (1,32,12,100) transpose doesn't work for -2, -3 dimensions. Namely, it throws an error that shape of the tensor must be divisible by tile dim. It seems that padding is not done in this case. 

As I looked into implementation of transpose I have noticed the difference: 
In `ttnn/cpp/ttnn/operations/data_movement/transpose/transpose.cpp` when padding is HC (-2,-3) padding is only done on C.

### **To Reproduce**

```
import pytest
import torch
import numpy as np

import ttnn

from loguru import logger
from models.utility_functions import is_grayskull
from tests.tt_eager.python_api_testing.sweep_tests.comparison_funcs import comp_pcc, comp_equal
from models.utility_functions import skip_for_grayskull, skip_for_blackhole
from tests.ttnn.utils_for_testing import assert_with_pcc


def test_transpose_hc(dtype, device):
    if is_grayskull() and dtype == ttnn.float32:
        pytest.skip("Skipping float32 tests on Grayskull")

    logger.info("transpose on C H dim")

    N = 1
    H = 32
    C = 12
    W = 100
    input_shape = (N, C, H, W)
    transpose(input_shape, device, dim0=1, dim1=-2, input_dtype=dtype)
```

tt-metal commit: 96fd0df449931e9c4958e761f40e0b8550c3e2c1

**Expected behaviour**
Transpose op should work regardless of the divisibility by tile dimensions.

To give a little context, we need this transpose case for Llama bringup on forge. This transpose case appears in self attention graph, and we need to support it.

**Comment by dgolubovicTT** [comment]
fyi @tt-mpantic

**Comment by nvukobratTT** [comment]
@dgolubovicTT for quicker and easier repro, can you provide the TTNN test code as part of the issue description ([example](https://github.com/tenstorrent/tt-metal/issues/12644)). 

Also, TTNN folks have their templates for opening up bug issues. In order to make their life a bit easier, let's try to use those template fields for information we currently posses (not all are mandatory).

**Comment by dgolubovicTT** [comment]
Hi @sjameelTT, did you have time to look into this one?

**Comment by sjameelTT** [comment]
Haven't had the chance yet. I'll get to it by the end of the week.

**Comment by sjameelTT** [comment]
I'm look at this right now. It's fixed on `sjameel/transpose_pad` atm but there's some other post-commit failures that I've introduced along the way so I'm working on that in the meantime.

**Comment by sjameelTT** [comment]
I'll try to get a PR in soon - it will be wrapped in with a bunch of other transpose changes to achieve better transpose coverage.

**Comment by sjameelTT** [comment]
It's in

**Comment by sjameelTT** [comment]
May I close this issue?

**Comment by dgolubovicTT** [comment]
If it is working on your side, yes. We will check from forge side as soon as we consume new metal.
